internal/netclassify/classify.go

```diff
@@ -36,6 +36,23 @@ var (
 	pythonSourceRegex = regexp.MustCompile(`^(?P<package>[\w\.]+)-(?P<version>.+?)(?P<ext>\.(zip|tar\.gz|tar\.bz2|tar\.xz|tar\.Z|tar))$`)
 )
 
+// NPM
+var (
+	npmAPIRegex  = regexp.MustCompile(`^https://registry\.(npmjs\.org|yarnpkg\.com)/((@[^/]+/)?[^/]+)/([^/]+)/?$`)
+	npmFileRegex = regexp.MustCompile(`^https://registry\.(npmjs\.org|yarnpkg\.com)/((@[^/]+/)?[^/]+)/-/([^/]+)-([^/-]+)\.tgz$`)
+)
+
+// Maven
+var (
+	mavenCentralRegex = regexp.MustCompile(`^https?://(repo1\.maven\.org/maven2|plugins.gradle.org/m2)/(.+)/([^/]+)/([^/]+)/([^/]+)$`)
```

Comment:
I don't know that this name is "correct", I think we need either a name that is more generic or we need two names here `mavenCentralRepo`, `gradlePluginRepo` with their own regexes?

Reply:
or mavenRepo? since the gradle plugin repo is strucured as a maven repository?

Reply:
yeah i think that's a good option. looks like the spec is here: https://cwiki.apache.org/confluence/display/MAVENOLD/Repository+Layout+-+Final